src/Microsoft.AspNetCore.Razor.OmniSharpPlugin.StrongNamed/OmniSharpWorkspaceProjectStateChangeDetector.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) .NET Fo    undation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+
+namespace Microsoft.AspNetCore.Razor.OmniSharpPlugin.StrongNamed
+{
+    public class OmniSharpWorkspaceProjectStateChangeDetector : OmniSharpProjectSnapshotManagerChangeTrigger
+    {
+        public OmniSharpWorkspaceProjectStateChangeDetector(
+            OmniSharpForegroundDispatcher foregroundDispatcher,
+            OmniSharpProjectWorkspaceStateGenerator workspaceStateGenerator)
+        {
+            if (foregroundDispatcher == null)
+            {
+                throw new ArgumentNullException(nameof(foregroundDispatcher));
+            }
+
+            if (workspaceStateGenerator == null)
+            {
+                throw new ArgumentNullException(nameof(workspaceStateGenerator));
+            }
+
+            InternalWorkspaceProjectStateChangeDetector = new ForegroundWorkspaceProjectStateChangeDetector(
```

Comment:
This is what does TagHelper resolution now. Trying to re-use what core-Razor uses so when we change one the other one gets updated as well. Basically this guy listens for `Workspace` & `ProjectSnapshotManager` change events and then decides if it wants to update TagHelpers. 

[Here's the Razor bits](https://github.com/aspnet/AspNetCore-Tooling/blob/8ee676d84cea1a375f6179aaf2db86f87f73625d/src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/WorkspaceProjectStateChangeDetector.cs).